### ISSUE TITLE
Runtime error - java.lang.ClassNotFoundException: Didn't find class "android.net.Network" in API level 19 with minifyEnabled is set to True

### ISSUE THREAD

**Issue by waresix** [transistorsoft/background-geolocation-lt#92]
Runtime error - java.lang.ClassNotFoundException: Didn't find class "android.net.Network" in API level 19 with minifyEnabled is set to True

## Your Environment
* Plugin version: 1.3.0
* Platform: Android
* OS version: Android 4.4.4 API 19
* Device manufacturer / model: Nexus 4
* Plugin config
config!!.updateWithBuilder()
    .setDesiredAccuracy(LocationRequest.PRIORITY_HIGH_ACCURACY)
    .setDistanceFilter(100F)
    .setStopTimeout(1L)
    .setHeartbeatInterval(180)
    .setStopOnTerminate(false)
    .setForegroundService(true)
    .setNotification(notif)
    .setStartOnBoot(true)
    .setUrl(RetrofitClient.GPS_URL)
    .setHeaders(headers)
    .setParams(params)
    .setAutoSync(true)
    .setEnableHeadless(true)
    .commit()

## Expected Behavior
Working normaly in Android 4.4.4 API 19 when minifyEnabled is set to true

## Actual Behavior
Transistorsoft crash when locationmanager is active in api level 19 (minifyEnabled set to True)

## Steps to Reproduce
1. Set minifyEnabled to True in release config
2. Run in Device with Android 4.4.4

## Context
Transistorsoft is working fine on API Level 21 and above when minifyEnabled is set to true, but in API level 19 the library crashed.
But if minifyEnabled is set to False, it's working even in API level 19. I already followed the proguard setting described in tutorial

## Debug logs
E/AndroidRuntime: FATAL EXCEPTION: pool-9-thread-1
    Process: com.waresix.freight.driver, PID: 25862
    java.lang.NoClassDefFoundError: android/net/Network
        at java.lang.Class.getDeclaredMethods(Native Method)
        at java.lang.Class.getPublicMethodsRecursive(Class.java:894)
        at java.lang.Class.getMethods(Class.java:877)
        at o0.b.a.l.a(SubscriberMethodFinder.java:38)
        at o0.b.a.l.a(SubscriberMethodFinder.java:24)
        at o0.b.a.c.c(EventBus.java:2)
        at com.transistorsoft.locationmanager.http.HttpService.<init>()
        at com.transistorsoft.locationmanager.http.HttpService.a()
        at com.transistorsoft.locationmanager.http.HttpService.getInstance()
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.<init>()
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.a()
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.getInstance()
        at com.transistorsoft.locationmanager.BootReceiver.a()
        at com.transistorsoft.locationmanager.BootReceiver.a()
        at com.transistorsoft.locationmanager.BootReceiver$a.run()
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "android.net.Network" on path: DexPathList[[zip file "/data/app/com.waresix.freight.driver-1.apk"],nativeLibraryDirectories=[/data/app-lib/com.waresix.freight.driver-1, /vendor/lib, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
        at java.lang.Class.getDeclaredMethods(Native Method) 
        at java.lang.Class.getPublicMethodsRecursive(Class.java:894) 
        at java.lang.Class.getMethods(Class.java:877) 
        at o0.b.a.l.a(SubscriberMethodFinder.java:38) 
        at o0.b.a.l.a(SubscriberMethodFinder.java:24) 
        at o0.b.a.c.c(EventBus.java:2) 
        at com.transistorsoft.locationmanager.http.HttpService.<init>() 
        at com.transistorsoft.locationmanager.http.HttpService.a() 
        at com.transistorsoft.locationmanager.http.HttpService.getInstance() 
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.<init>() 
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.a() 
        at com.transistorsoft.locationmanager.adapter.BackgroundGeolocation.getInstance() 
        at com.transistorsoft.locationmanager.BootReceiver.a() 
        at com.transistorsoft.locationmanager.BootReceiver.a() 
        at com.transistorsoft.locationmanager.BootReceiver$a.run() 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:841)

**Comment by christocracy** [comment]
See required [Proguard Config](https://github.com/transistorsoft/background-geolocation-lt/blob/master/docs/INSTALL-ANDROID.md#proguard-config) in *Android Setup*.

**Comment by waresix** [comment]
I have done that, but still the same. Is there any workaround? This is only happening in API 19.

```
buildTypes {
    release {
        minifyEnabled true
        proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        proguardFiles "${background_geolocation.projectDir}/proguard-rules.pro"
    }
}
```

**Comment by christocracy** [comment]
One of my 11 test devices is a *Nexus 4 @ 4.4.4*.  It has no issue.  

I don't know what your issue could be.

Perhaps you need to implement [MultiDex](https://developer.android.com/studio/build/multidex)?

**Comment by waresix** [comment]
May I know your Android Gradle Plugin Version and Gradle version?
Thank you.

This is my Gradle version.
![image](https://user-images.githubusercontent.com/50509795/84721730-e4272980-afb3-11ea-99c2-62136fa0d616.png)

**Comment by arifmc** [comment]
Experiencing the same issue, any solution?

**Comment by christocracy** [comment]
@arifmc ”me too” responses are useless.  You need to provide all the same information as OP:  

**edit** your previous post with *information*.

- Plugin version: 
- Platform: 
- OS version:
- Device manufacturer / model:
- Plugin config

**Comment by christocracy** [comment]
@waresix 
- *Android Gradle Plugin Version*:  `3.4.2`
- *Gradle Version*: `5.1.1`

**Comment by christocracy** [comment]
I've just released [1.4.0](https://github.com/transistorsoft/background-geolocation-lt/pull/93).  

Take note of new **required Android Setup Step** in the root `android/build.gradle`.

This is also works with
- *Android Gradle Plugin Version*: `4.0.0`
- *Gradle Version: `6.1.1`

**Comment by waresix** [comment]
I have updated the version according to the information and it works well now.
Thank you.
